src/analysis/seminorm.lean

```diff
@@ -334,11 +375,28 @@ lemma mem_ball_zero : y ∈ ball p 0 r ↔ p y < r := by rw [mem_ball, sub_zero]
 
 lemma ball_zero_eq : ball p 0 r = { y : E | p y < r } := set.ext $ λ x, p.mem_ball_zero
 
+@[simp] lemma ball_zero' (x : E) (hr : 0 < r) : ball (0 : seminorm 𝕜 E) x r = set.univ :=
+begin
+  rw [set.eq_univ_iff_forall, ball],
+  simp [hr],
+end
+
+lemma ball_sup (p : seminorm 𝕜 E) (q : seminorm 𝕜 E) (e : E) {r : ℝ} (hr : 0 < r) :
+  ball (p ⊔ q) e r = ball p e r ∩ ball q e r :=
+begin
+  lift r to nnreal using hr.le,
+  simp_rw [ball, ←set.set_of_and],
+  simp,
+end
+
```

Comment:
And another lemma...
```lean

lemma ball_finset_sup' (p : ι → seminorm 𝕜 E) (s : finset ι) (H : s.nonempty) (e : E) (r : ℝ) :
  ball (s.sup' H p) e r = s.inf' H (λ i, ball (p i) e r) :=
begin
  induction s using finset.cons_induction_on with a s ha ih,
  { exact false.elim (not_nonempty_empty H), },
  { rcases s.eq_empty_or_nonempty with rfl | hs,
    { classical, simp },
    { rw [finset.sup'_cons hs, finset.inf'_cons hs, ball_sup, inf_eq_inter, ih] } },
end

```
Note that this one also does not put any requirements on `r`

Reply:
that code fails for me with 'couldn't resolve `0 ≤ r`'

Reply:
Can you paste the full error? It worked for me, but vscode might have been misbehaving

Reply:
Oh, you need my suggestion above too